token/oauth/oauth2.go

```diff
@@ -101,10 +101,18 @@ func SaveReferrer(ctx context.Context, db application.DB, state string, referrer
 	}
 	// TODO The state reference table will be collecting dead states left from some failed login attempts.
 	// We need to clean up the old states from time to time.
-	ref := auth.OauthStateReference{
-		State:        state,
-		Referrer:     referrer,
-		ResponseMode: responseMode,
+	var ref auth.OauthStateReference
+	if responseMode != nil {
```

Comment:
Just define OauthStateReference.ResponseMode as *string instead of string. Then you don't have to check if it's nil or not here.

Reply:
OauthStateReference is stored in the database, if I change it to *string, I think it will try to save a pointer in the database and not the value.

Reply:
No. If you change it to *string then you will be able to store NULL or a string value in DB.
If you keep it as string then instead of NULL you will store an empty string.